src/schema/updates/manifest.json

```diff
@@ -45,6 +45,15 @@
                       ]
                     },
                     "uniqueItems": true
+                  } ,
+                  "optioanl_permissions": {
+                    "items": {
+                      "anyOf": [
+                        {},
+                        { "format": "deprecated" }
```

Comment:
@aa-deet-eeya there is a typo here at line 49: "optioanl_permissions" -> ""optional_permissions"

I did some digging into the git history of this file to double-check what is the actual purpose of `anyOf` part, but unfortunately the pull request that originally added it doesn't provide any additional context about that.
To be honest I'm pretty sure that the `"format": "deprecated"` part of it isn't actually used (and removing that part from "permissions" doesn't actually trigger any test failure), and so at least at a first glance it seems that it is not that useful (even on "permissions" where it is applied from a long long time).

Anyway, I will keep it for now, for consistency between "permissions" and "optional_permissions" (and we will eventually evaluate if we could/should remove it in a separate issue).

I did run the schema import script locally to double-check that the change applied to src/schema/imports/manifest.json because of this additional annotation are all applied as expected (otherwise we would have some unrelated changes in the next schema import after merging this on master), and we would need to apply the following changes to src/schema/imported/manifest.json:

```diff
diff --git a/src/schema/imported/manifest.json b/src/schema/imported/manifest.json
index 8b274bcb..16356cc8 100644
--- a/src/schema/imported/manifest.json
+++ b/src/schema/imported/manifest.json
@@ -241,7 +241,6 @@
                 },
                 "optional_permissions": {
                   "type": "array",
-                  "default": [],
                   "items": {
                     "allOf": [
                       {
@@ -250,8 +249,15 @@
                       {
                         "onError": "warn"
                       }
+                    ],
+                    "anyOf": [
+                      {},
+                      {
+                        "format": "deprecated"
+                      }
                     ]
                   },
+                  "default": [],
                   "uniqueItems": true
                 },
                 "web_accessible_resources": {
```

would you mind to also apply this changes?

In the meantime I'll give another review round to the rest of the changes in this pull request.